app/controllers/RestaurantController.java

```diff
@@ -101,4 +103,15 @@ public Result createRestaurant(){
         PersistenceManager.saveRestaurant(restaurant);
         return created(JsonSerializer.serializeObject(new RestaurantHelper(restaurant.getRestaurantId())));
     }
+
+    @Security.Authenticated(UserAuthenticator.class)
+    public Result makeFavourite(int restaurantId){
+        User user =(User) Http.Context.current().args.get("CurrentUser");
+        Restaurant restaurant = PersistenceManager.getRestaurantById(restaurantId);
+        if(restaurant == null) {
+            return badRequest(JsonSerializer.serializeObject(new Error(Resources.NO_RESTAURANT)));
+        }
+        PersistenceManager.makeFavourite(user, restaurant);
```

Comment:
I don't see point of passing user to persistence layer. You can get user from any part of code (preferably with UserHelper.getCurrentUser() ).